baton_app/baton_app.c

```diff
@@ -387,3 +409,130 @@ int baton_client_loop_cb(picoquic_quic_t* quic, picoquic_packet_loop_cb_enum cb_
     }
     return ret;
 }
+
+/* Sample server.
+* The baton server is an HTTP3 server.
+* It can server web pages like a regular server, but will
+* also respond to a webtransport "connect" request and start
+* a baton connection.
+* The server parameters are set using the configuration
+* arguments standard for the picoquic library.
+*/
+char const* baton_ticket_store = "baton_ticket_store.bin";
+char const* baton_token_store = "baton_token_store.bin";
+
+int wt_baton_server(char const* path, picoquic_quic_config_t* config)
+{
+    int ret = 0;
+    picohttp_server_path_item_t path_item_list[1] = { { 0 } };
+
+    /* Start: start the QUIC process with cert and key files */
+    picoquic_quic_t* qserver = NULL;
+    uint64_t current_time = 0;
+    picohttp_server_parameters_t picoquic_file_param;
+
+    /* set the transport parameters to a proper value */
+    memset(&picoquic_file_param, 0, sizeof(picohttp_server_parameters_t));
+
+    /* Configure the web server parameters, including the "path table"
+    * with a path item describing the baton app.
+    * The "path_app_context" is null because the baton app does not use it.
+     */
+    picoquic_file_param.web_folder = config->www_dir;
+    path_item_list[0].path = path;
+    path_item_list[0].path_length = strlen(path);
+    path_item_list[0].path_callback = wt_baton_callback;
```

Comment:
Should we check if the service was indeed baton here before assinign the path_callback ?